src/runtime/TypeManager.cs

```diff
@@ -587,7 +587,7 @@ internal static PyType AllocateTypeObject(string name, PyType metatype)
             // Clr type would not use __slots__,
             // and the PyMemberDef after PyHeapTypeObject will have other uses(e.g. type handle),
             // thus set the ob_size to 0 for avoiding slots iterations.
-            Util.WriteIntPtr(type, TypeOffset.ob_size, IntPtr.Zero);
+            Util.WriteCLong(type, TypeOffset.ob_size, 0);
```

Comment:
This looks wrong to me. ob_size is `Py_ssize_t` which AFAIK is explicitly defined to match pointer sizes.